apps/server/src/modules/tldraw/service/tldraw-files-storage.service.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@nestjs/common';
+import { FileDto, FilesStorageClientAdapterService } from '@modules/files-storage-client';
+import { TldrawAsset } from '../types';
+
+@Injectable()
+export class TldrawFilesStorageAdapterService {
+	constructor(private readonly filesStorageClientAdapterService: FilesStorageClientAdapterService) {}
+
+	public async deleteUnusedFilesForDocument(docName: string, usedAssets: TldrawAsset[]) {
+		const deleteFilePromises: Promise<FileDto>[] = [];
+		const fileRecords = await this.filesStorageClientAdapterService.listFilesOfParent(docName);
+		for (const fileRecord of fileRecords) {
+			const foundAsset = usedAssets.find((asset) => {
+				const srcArr = asset.src.split('/');
+				const fileRecordId = srcArr[srcArr.length - 2];
+
+				return fileRecordId === fileRecord.id;
+			});
+
+			if (!foundAsset) {
+				deleteFilePromises.push(this.filesStorageClientAdapterService.deleteOneFile(fileRecord.id));
+			}
+		}
```

Comment:
With this change the deleteUnusedFilesForDocument only include orechstration and in XXXX the logic is placed. You can also use array.reduce if you like it more.  The XXXX is maybe something like "createFileDeletionActionWhenAssetNotExists" .

```
const deleteFilePromises = fileRecords.map((fileRecord) => this.XXXX(fileRecord, usedAssets));

private XXXX(fileRecord, usedAssets) {
    const foundAssets = usedAssets.find((asset) => this.matchAssetWithFileRecord(asset, fileRecord));
    const promise = foundAsset ? Promise.resolve() :  this.filesStorageClientAdapterService.deleteOneFile(fileRecord.id);
   
    return promise ;
}
```

Reply:
Good idea, refactored to change deleteUnusedFilesForDocument into orchestrator and moved logic into private methods.